### ISSUE TITLE
Cannot read properties of undefined (reading 'valueX') when using snapTooltip

### ISSUE THREAD

**Issue by samkristoff** [amcharts/amcharts5#49]
Hey, 

I have an XY chart with a DateAxis (x) and Value Axis (y).  I enabled **panX** on the chart and set a large **maxDeviation** on the x axis so that the user can pan left and right beyond the edge of the existing data.  I setup a 'debounce' timer to detect when the user has stopped panning (ie 500ms since the last pan) and then fetch more series data and update the chart.

```
...

// Create chart
this.chart = this.root.container.children.push(
  am5xy.XYChart.new(this.root, {
    panX: true,
    panY: false,
    wheelY: "zoomX",
  })
);

// Add x (date) axis
this.xAxis = this.chart.xAxes.push(
  am5xy.DateAxis.new(this.root, {
    renderer: am5xy.AxisRendererX.new(this.root, {}),
    maxDeviation: 4, // Allows user to zoom out beyond data (as % of data width ex. 4 = 400%)
    baseInterval: {
      count: 1,
      timeUnit: "second",
    },
  })
);

...
```

This works well when **snapTooltip** is not enabled, but when I enable tooltips I often see the following error when panning.
![image](https://user-images.githubusercontent.com/1994983/139726958-dcc5f026-a896-455e-9bb6-1e088192eb3b.png)

```
this.series[measureName] = this.chart.series.push(
  am5xy.LineSeries.new(this.root, {
    name: measureName,
    xAxis: this.xAxis,
    yAxis: yAxis,
    valueYField: "val",
    valueXField: "ts",
    snapTooltip: true,  // <-- After commenting this the error no longer presents itself
    tooltip: am5.Tooltip.new(this.root, {}),
  })
);
```

From what I can tell this is coming from inside amcharts somewhere and I can't 'catch' the error.  Please let me know if there is some additional data I can provide to help debug this.

**Comment by samkristoff** [comment]
I added a 'refresh' button so I can re-load a data set without panning/zooming the chart and make sure I'm loading the same data each time.   

Clicking the refresh button does not throw the error above if I keep the cursor outside the chart.  However if I click the refresh button and move my cursor around over the chart the error is thrown.

I save a reference to each series in a map (this.series) the first time I add it to the chart and then update the data with 

```
this.series[seriesName].data.setAll(newData);
```

**Comment by zeroin** [comment]
It would be best if you could create a codepen where this issue is reproduced, it would help to figure out the problem - thanks!

**Comment by samkristoff** [comment]
I was able to reproduce this in JSFiddle using a setTimeout to mock the API call.  
https://jsfiddle.net/samkristofftulu/qwrgx7mv/65/

Here is a video demonstrating the issue.
https://share.vidyard.com/watch/LJo4navHWFRv1XU2gsRYsN?

**Comment by zeroin** [comment]
I confirm the bug - we fixed it internally and the issue will be solved with next release. Thanks a lot for helping to find it.

**Comment by samkristoff** [comment]
Hey zerio, I'm happy to help.

Is there a work around or patch I can try to see if this also prevents my charts from locking up (not sure if this is just a red herring).

Thanks!

**Comment by martynasma** [comment]
BTW, we'll let you know here when the version with a fix is available.

**Comment by martynasma** [comment]
Fixed in [5.0.1](https://github.com/amcharts/amcharts5/releases/tag/5.0.1).

## [5.0.1] - 2021-11-05

### Fixed
- `XYChart` with panning enabled an no Animated theme was flickering while panning.
- JS error could occur sometimes if data was updated while cursor was moving over the chart.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download-v5/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
